ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,669 @@
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_busy = true;
+  emit isBusy();
+  updateMessage("Loading a webmap...");
+
+  const Envelope envelope(
+      -9815489.0660101417,
+      5128463.4221229386,
+      -9814625.2768726498,
+      5128968.4911854975,
+      SpatialReference::webMercator()
+      );
+
+  Viewpoint initialViewpoint(envelope);
+
+  m_map->setInitialViewpoint(initialViewpoint);
+
+  // Load in persistent session mode (workaround for server caching issue)
+  // https://support.esri.com/en-us/bug/asynchronous-validate-request-for-utility-network-servi-bug-000160443
+  m_map->loadSettings()->setFeatureServiceSessionType(FeatureServiceSessionType::Persistent);
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  GraphicsOverlay* m_overlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(m_overlay);
+
+  emit mapViewChanged();
+
+  connectSignals();
+}
+
+void ValidateUtilityNetworkTopology::onMapLoaded()
+{
+  // Load and switch utility network version
+  updateMessage("Loading the utility network...");
+
+  m_utilityNetwork = m_map->utilityNetworks()->first();
+  m_utilityNetwork->load();
+
+  m_serviceGeodatabase = m_utilityNetwork->serviceGeodatabase();
+  const QString randomVersionUuid = QUuid().createUuid().toString(QUuid::WithoutBraces);
+
+  // Restrict editing and tracing on a random branch
+  ServiceVersionParameters* params = new ServiceVersionParameters(this);
+  params->setName("ValidateNetworkTopology_" + randomVersionUuid);
+  params->setAccess(VersionAccess::Private);
+  params->setDescription("Validate network topology with ArcGIS Runtime");
+
+  connect(m_serviceGeodatabase, &ServiceGeodatabase::doneLoading, this, [this, params](const Esri::ArcGISRuntime::Error &loadError)
+          {
+            onServiceGeodatabaseLoaded(params, loadError);
+          });
+
+  connect(m_utilityNetwork, &UtilityNetwork::doneLoading, this, [this]()
+          {
+            setupTraceParameters();
+          });
+}
+
+void ValidateUtilityNetworkTopology::onServiceGeodatabaseLoaded(ServiceVersionParameters* params, const Error& loadError)
+{
+  if (!loadError.isEmpty())
+    return;
+
+  m_serviceGeodatabase->createVersionAsync(params).then(this, [this](ServiceVersionInfo* serviceVersionInfo)
+  {
+    m_serviceGeodatabase->switchVersionAsync(serviceVersionInfo->name()).then(this, [this]()
+    {
+      displayLabelDefinitions();
+
+      // Visualize dirty area by adding to the map
+      ArcGISFeatureTable* m_dirtyAreaTable = m_utilityNetwork->dirtyAreaTable();
+      m_map->operationalLayers()->append(new FeatureLayer(dynamic_cast<FeatureTable*>(m_dirtyAreaTable), this));
+    });
+  });
+}
+
+void ValidateUtilityNetworkTopology::connectSignals()
+{
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& mouseEvent)
+  {
+    // first clear the selection
+    m_deviceFeatureLayer->clearSelection();
+    m_lineFeatureLayer->clearSelection();
```

Comment:
we need to be careful when accessing members without a null check. I tested the branch locally, and got a crash right away. I believe it is because I tried clicking on a feature on the map before the map/utility network were fully loaded, and something was in a null state.